code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -273,36 +276,61 @@ GLOBAL_DATUM_INIT(pipe_icon_manager, /datum/pipe_icon_manager, new())
 
 	var/obj/machinery/atmospherics/target_move = findConnecting(direction)
 	if(target_move)
-		if(is_type_in_list(target_move, GLOB.ventcrawl_machinery) && target_move.can_crawl_through())
-			user.remove_ventcrawl()
-			user.forceMove(target_move.loc) //handles entering and so on
-			user.visible_message("You hear something squeezing through the ducts.", "You climb out the ventilation system.")
+		if(is_type_in_list(target_move, GLOB.ventcrawl_machinery))
+			if(target_move.can_crawl_through())
+				if(do_after(user, 30, target = target_move))
+					user.remove_ventcrawl()
+					user.forceMove(target_move.loc) //handles entering and so on
+					user.visible_message("You hear something squeezing through the ducts.", "You climb out the ventilation system.")
+			else if(istype(user, /mob/living/silicon/robot/drone))
+				var/mob/living/silicon/robot/drone/D = user
+				if(D.emagged)
+					return
+				to_chat(user, "<span class='notice'>Using specialized micro tools you begin disconnecting the [target_move] from its frame....</span>")
+				if(do_after(user, 150, target = target_move))
+					user.remove_ventcrawl()
+					user.forceMove(target_move.loc) //handles entering and so on
+					user.visible_message("You hear something squeezing through the ducts. With a resounding snap the [target_move] is fastened back in place.", \
+						"You climb out the ventilation system. With a resounding snap the [target_move] is fastened back in place.")
```

Comment:
While you're here, could you add opening and closing spans on any `visible_message`'s and `to_chat`'s in areas of code you've touched?